src/code/fault_ootn64.c

```diff
@@ -0,0 +1,835 @@
+    0x08888800, 0x09900000, 0x00111000, 0x09922440, 0x00011000, 0x09908800, 0x26EFDE20, 0x099BB540, 0x2EC33CE2,
+    0x0D9A2550, 0x2EC33CE2, 0x0DDAA550, 0x2EC33CE2, 0x09D6ED10, 0x26CBBC62, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00011000, 0x05FBFFE0, 0x8E6116E8, 0x0FF40330, 0x8F7117F8, 0x07FC8B30, 0x8E6996E8,
+    0x05733BA0, 0x8A6DD6A8, 0x0DD88A20, 0x08A779B2, 0x01100220, 0x00000000, 0x00000080, 0x8A011000, 0x00000800,
+    0x80A11000, 0x07744F70, 0x80A99000, 0x0231DF20, 0x84E60004, 0x0027DA20, 0xC8AA4C40, 0x00573B20, 0x00A11800,
+    0x05546F50, 0x00A99800, 0x02222080, 0x02001888,
+};
+
+const char* sExceptionNames[] = {
+    "Interrupt",
+    "TLB modification",
+    "TLB exception on load",
+    "TLB exception on store",
+    "Address error on load",
+    "Address error on store",
+    "Bus error on inst.",
+    "Bus error on data",
+    "System call exception",
+    "Breakpoint exception",
+    "Reserved instruction",
+    "Coprocessor unusable",
+    "Arithmetic overflow",
+    "Trap exception",
+    "Virtual coherency on inst.",
+    "Floating point exception",
+    "Watchpoint exception",
+    "Virtual coherency on data",
+};
+
+const char* sFpExceptionNames[] = {
+    "Unimplemented operation", "Invalid operation", "Division by zero", "Overflow", "Underflow", "Inexact operation",
+};
+
+u16 sFaultFontColor = GPACK_RGBA5551(255, 255, 255, 1);
+
+Input sFaultInputs[MAXCONTROLLERS];
+
+FaultMgr gFaultMgr;
+
+STACK(sFaultStack, 0x400);
+StackEntry sFaultStackInfo;
+struct FaultCursorCoords sFaultCursorPos;
+
+vs32 sFaultExit;
+vs32 gFaultMsgId;
+vs32 sFaultDisplayEnable;
+OSThread* sFaultFaultedThread;
+s32 B_80122570[0x10];
+
+void Fault_SleepImpl(u32 ms) {
+    Sleep_Msec(ms);
+}
+
+void Fault_WaitInputImpl(void) {
+    Input* inputs = sFaultInputs;
+    u16 btnPress;
+
+    do {
+        Fault_SleepImpl(0x10);
+        PadMgr_RequestPadData(&gPadMgr, inputs, 0);
+        btnPress = inputs[0].press.button;
+    } while (!CHECK_BTN_ANY(btnPress, (BTN_A | BTN_B | BTN_START | BTN_CRIGHT | BTN_CLEFT | BTN_CDOWN | BTN_CUP)));
+}
+
+void Fault_WaitInput(void) {
+    Fault_WaitInputImpl();
+}
+
+void Fault_DrawRec(s32 x, s32 y, s32 w, s32 h, u16 color) {
+    s32 temp_a3;
+    s32 i;
+    s32 j;
+    u16* fbPtr;
+
+    fbPtr = (void*)((gFaultMgr.fbWidth * y * 2) + gFaultMgr.framebuffer + (x * 2));
+
+    for (i = 0; i < h; i++) {
+        j = 0;
+        for (j = 0; j < w; j++) {
+            *fbPtr = color;
+            fbPtr++;
+        }
+        fbPtr += (gFaultMgr.fbWidth - w);
+    }
+    osWritebackDCacheAll();
+}
+
+void Fault_DrawRecBlack(s32 x, s32 y, s32 w, s32 h) {
+    Fault_DrawRec(x, y, w, h, GPACK_RGBA5551(0, 0, 0, 1));
+}
+
+void Fault_DrawCharImpl(s32 x, s32 y, char c) {
+    u32* dataPtr;
+    s32 shift = c % 4;
+    s32 i;
+    s32 j;
+    u16* fbPtr;
+    u32 data;
+
+    dataPtr = (u32*)sFaultDrawerFont + ((c / 8 * 0x10) + ((c & 4) >> 2));
```

Comment:
`sFaultDrawerFont[...]`?

Reply:
doesn't match, at least not out of the box. leaving it to someone else